app/src/main/java/org/oppia/android/app/recyclerview/BindableAdapter.kt

```diff
@@ -99,34 +99,13 @@ class BindableAdapter<T : Any> internal constructor(
    * The base builder for [BindableAdapter]. This class should not be used directly--use either
    * [SingleTypeBuilder] or [MultiTypeBuilder] instead.
    */
-  abstract class BaseBuilder<BuilderType : Any> {
+  abstract class BaseBuilder(fragment: Fragment) {
     /**
-     * A [WeakReference] to a [LifecycleOwner] for databinding inflation. See [setLifecycleOwner].
+     * A [WeakReference] to a [LifecycleOwner] for databinding inflation.
      * Note that this needs to be a weak reference so that long-held references to the adapter do
      * not potentially leak lifecycle owners (such as fragments and activities).
      */
-    private var lifecycleOwnerRef: WeakReference<LifecycleOwner>? = null
-
-    /**
-     * Sets the [LifecycleOwner] corresponding to this adapter. This will automatically be used as
-     * the lifecycle owner for all databinding classes created during view inflation. Note that the
-     * adapter holds a weak reference to the owner to ensure long-lived references to the adapter
-     * class itself does not result in leaks, however it's up to the caller's responsibility to make
-     * sure that the adapter itself is not actually used after the lifecycle owner has expired
-     * (otherwise the app may crash).
-     *
-     * @return this
-     */
-    fun setLifecycleOwner(lifecycleOwner: LifecycleOwner): BuilderType {
-      check(lifecycleOwnerRef == null) {
-        "A lifecycle owner has already been bound to this adapter."
-      }
-      lifecycleOwnerRef = WeakReference(lifecycleOwner)
-
-      // This cast is not, strictly speaking, safe, but child classes are expected to inherit from
-      // the builder & pass their own type in.
-      @Suppress("UNCHECKED_CAST") return this as BuilderType
-    }
+    private var lifecycleOwnerRef: WeakReference<LifecycleOwner> = WeakReference(fragment)
```

Comment:
Can be a val, I think, since it'll never change.

Reply:
Yeah for sure